meta/Search.php

```diff
@@ -342,7 +358,8 @@ public function getSQL() {
             } else {
                 // first table
 
-                if(!$schema->isLookup()) {
+                // FIXME this breaks page search, a different check is needed
+                if(false) {
```

Comment:
Ohhh. I see what you mean. It's similar to the problem further up at https://github.com/cosmocode/dokuwiki-plugin-struct/pull/489/files#r397089182

I think it should be possible to always add these where clauses but wrap them in parentheses and OR them together with a check on `pid  IS NULL` .

Eg. `WHERE (pid IS NULL OR (GETACCESSLEVEL(pid) > 0 AND PAGEXISTS(pid) =1 AND ....)`

Reply:
Maybe my implementation was not clean enough but the suggested solution added unwanted lookups to page aggregations. Because basically it translates into "not page data OR data associated with accessible pages".

I think at this point the new `$idColumn === 'pid'` (join column) is a decent stand-in for `$schema->islookup()` and I'm using it instead in the checks. Unit and manual test look good so far.